Makefile

```diff
@@ -166,11 +167,20 @@ test-single-integration:
 	TEST_NATIVE_POOLING=false make test-base-single-integration name=$(name)
 
 .PHONY: test-ci-unit
-test-ci-unit: test-base-ci-unit
+test-ci-unit: test-base
+	cat $(coverfile) | egrep -v -f $(coverage_exclude) > tmp_$(coverfile) && mv tmp_$(coverfile) $(coverfile)
+	bash <(curl -s https://codecov.io/bash) -f $(coverfile) -F unittests
```

Comment:
doesn't seem much worse than `go get <random_go_tool` and then running that

Reply:
Yeah I was in two minds about doing this. It's a terrible practice in general - I refuse to run scripts like this on my laptop. It's much worse than go get (e.g. [hijack](https://www.idontplaydarts.com/2016/04/detecting-curl-pipe-bash-server-side/)). I put it in cause it's only going to run on the ci servers but happy to change to avoid doing in general.

Reply:
put the bash script in `m3db/ci-scripts`